core.go

```diff
@@ -351,6 +365,34 @@ func (sock *socket) SetOption(name string, value interface{}) error {
 		sock.wdeadline = value.(time.Time)
 		sock.Unlock()
 		return nil
+	case OptionWriteQLen:
+		sock.Lock()
+		defer sock.Unlock()
+		active := sock.active
```

Comment:
I don't think the scratch variable here (active) adds any value... just do if sock.active { ...